### PR TITLE
GOBBLIN-1352: Ensure Helix workflows are cleaned up on cluster start up in centralized mode

### PR DESCRIPTION
**Pull request by sv2000**
Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [x] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title. For example, "[GOBBLIN-XXX] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-1352

### Description
- [x] Here are some details about my PR, including screenshots (if applicable):
As part of the changes introduced in GOBBLIN-1319, the Helix workflow names were modified to include a special prefix. However, this prefix is intended for use in distributed mode. As a consequence of this change, the workflows are no longer being cleaned up on cluster start up leading to a leak in Znodes. Further, since the workflows are left in the START state, the Yarn autoscaling manager requests more containers than needed.

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
Added unit test.

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

**Comment by sv2000** [comment]
@autumnust @ZihanLi58 Please review.

File: gobblin-cluster/src/test/java/org/apache/gobblin/cluster/HelixRetriggeringJobCallableTest.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  }
+
+  @Test
+  public void testBuildJobLauncher()
+      throws Exception {
+    Config config = ConfigFactory.empty().withValue(ConfigurationKeys.JOB_CONFIG_FILE_GENERAL_PATH_KEY,
+        ConfigValueFactory.fromAnyRef(TMP_DIR));
+    MutableJobCatalog jobCatalog = new NonObservingFSJobCatalog(config);
+    SchedulerService schedulerService = new SchedulerService(new Properties());
+    //HelixManager helixManager = Mockito.mock(HelixManager.class);
```
Comment (autumnust) [inline gobblin-cluster/src/test/java/org/apache/gobblin/cluster/HelixRetriggeringJobCallableTest.java:61]: Shall we remove these lines ?

**Review by autumnust — APPROVED** [review]
+1

**Review by autumnust — APPROVED** [review]
LGTM

**Review by autumnust — APPROVED** [review]
+1

**Pull request closed without merging**